rand_pcg/Cargo.toml

```diff
@@ -25,8 +25,11 @@ serde1 = ["serde", "serde_derive", "bincode/i128"]
 rand_core = { version = "0.3", default-features=false }
 serde = { version = "1", optional = true }
 serde_derive = { version = "^1.0.38", optional = true }
-# required for testing Serde, and we must depend on it to request i128 support
-bincode = { version = "1", optional = true }
+
+[dev-dependencies]
+# This is for testing serde, unfortunately we can't specify feature-gated dev
+# deps yet, see: https://github.com/rust-lang/cargo/issues/1596
+bincode = "1"
```

Comment:
Hm, isn't having `bincode/i128` enough? Serde tests are passing without any problems, or am I missing something?

Reply:
Yes, that's what happened to me first time. `dev-dependencies` are used during testing but not from external crates, hence if another crate depends on one of these with the `serde1` feature Cargo complains that this crate doesn't have a `bincode` feature.

Reply:
One possible alternative is to remove `bincode/i128`, specify dependency as:
```toml
[dev-dependencies]
bincode = { version = "1", features = ["i128"] }
```
And use `cargo build` instead of `cargo test` for Minimum Supported Rust Version test. It's not ideal, but I think better than having `bincode` in direct dependencies.

Reply:
Sorry, that's not an alternative because `bincode` needs that feature specified for compatibility with *modern* Rustc versions. It's really a failing on their part, hence the issue I linked above.

Since Serde needs `bincode` I don't see any real harm in directly depending on it. We could instead do the following but I don't think it makes any difference (from the current version, which is what the last release uses):
```toml
[features]
serde1 = ["serde", "serde_derive", "bincode"]

[dependencies]
bincode = { version = "1", features = ["i128"], optional = true }
```

Reply:
Shouldn't this `bincode` feature be enabled by users who will us it for serialization? It's just strange that PRNG implementation depends on serialization format crate. Say I want to serialize PCG to JSON, why should I have bincode in my dependency tree?

Reply:
No idea.

So should we just tell users: *if you use Serde with binary encoding, you will need to depend on `bincode` with the `i128` feature*?

I suppose we can do that, and point anyone who complains to the above bug report. You're right, it's not really our problem.

Reply:
Yeah, I think that documenting it will be the best solution for now. I will try to add it a bit later.